MREUnityRuntime/MREUnityRuntimeLib/Core/Collider.cs

```diff
@@ -30,56 +30,78 @@ public enum ColliderType
         /// </summary>
         None = 0,
 
+        /// <summary>
+        /// Choose best collider shape for mesh
+        /// </summary>
+        Auto,
+
         /// <summary>
         /// Box shaped collider.
         /// </summary>
-        Box = 1,
+        Box,
 
         /// <summary>
         /// Sphere shaped collider.
         /// </summary>
-        Sphere = 2,
+        Sphere,
 
         /// <summary>
-        /// Mesh based collider.
+        /// Capsule shaped collider.
         /// </summary>
-        Mesh = 3,
+        Capsule,
 
         /// <summary>
-        /// Capsule shaped collider.
+        /// Mesh collider.
         /// </summary>
-        Capsule = 4
+        Mesh
     }
 
     internal class Collider : MonoBehaviour, ICollider
     {
         private UnityCollider _collider;
-        private Actor _ownerActor;
         private ColliderEventType _colliderEventSubscriptions = ColliderEventType.None;
+        private int colliderGeneration = -1;
+
+        private Actor _actor;
+        private Actor Actor
+        {
+            get
+            {
+                if (_actor == null)
+                {
+                    _actor = gameObject.GetComponent<Actor>();
+                }
+                return _actor;
+            }
+        }
 
         /// <inheritdoc />
-        public bool IsEnabled => _collider.enabled;
+        public bool IsEnabled { get; private set; } = true;
 
         /// <inheritdoc />
-        public bool IsTrigger => _collider.isTrigger;
+        public bool IsTrigger { get; private set; } = false;
```

Comment:
I don't agree with this change.  The actual collider is the authority of these values, not our wrapper class here.